File: Backend/MuseumAPI/Models/Museum.cs
using Azure;

namespace MuseumAPI.Models
{
    public class Museum
    {
        public long Id { get; set; }
        public string? Name { get; set; }
        public string? Address { get; set; }
        public DateTime? FoundationDate { get; set; }
        public string? Architect { get; set; }
        public string? Website { get; set; }

        // Hidden from the API because it's not in the DTO
        public virtual ICollection<Artist> Artists { get; set; } = null!;
        public virtual ICollection<Exhibition> Exhibitions { get; set; } = null!;

        public virtual long? UserId { get; set; }
        public virtual User? User { get; set; } = null!;
    }
}

File: Frontend/src/models/Museum.ts
import { Artist } from './Artist';
import { Exhibition } from './Exhibition';
import { User } from './User';

export interface Museum {
    id?: number;
    name: string;
    address: string;
    foundationDate: Date;
    architect: string;
    website: string;
    artists?: Artist[];
    exhibitions?: Exhibition[];

    userId?: number;
    user?: User;

    [key: string]: any;
}